controllers/nodemaintenance_controller.go

```diff
@@ -339,7 +373,7 @@ func (r *NodeMaintenanceReconciler) stopNodeMaintenanceOnDeletion(ctx context.Co
 			if err := r.LeaseManager.InvalidateLease(ctx, &corev1.Node{ObjectMeta: metav1.ObjectMeta{Name: nodeName}}); err != nil {
 				return err
 			}
-			return nil
+			return r.removeExcludeRemediationLabel(ctx, node)
```

Comment:
this doesn't make sense here, this code branch runs when we don't have a node anymore 🙂

Reply:
Good catch ! I was following the lease trail and missed the context